### PR TITLE
✨🎨 Implementing a Video Upload Modal when a user clicks on the play button #336

### LINKED ISSUE #336: 🗒️ Video Block Modal
## User stories

As a user I would like to see a modal similar to the one below when I attempt to attach a video so that I can be able to choose the size of the video I would like to upload.

---

Quick links: [Summary](#summary) | [Description](#description) | [Template](#template) | [Example](#example) | [Resources](#resources)

---

### Description
Currently how the video block works, is once the play icon is clicked it opens the file explorer to allow  the upload the video file.

In this feature instead of opening the file explorer on click, a modal with the design below should be implemented and open the file explorer when the thumbnail space is clicked.

```markdown
1. Navigate to the video block
1. I click on the play icon.
1. A modal window should appear
1. The modal window contains the following:
  1. [A title]
  2. [Two input fields: One a name input and Second a video input]
  3. [Radio button options with the size options for the video **Note** : One should be able to select only one option]
1. A button that applies the picked option, the video file and the name of the video to the video block object.
```

**Screenshots**

Screenshot 2023-03-01 at 16 19 47

### Acceptance Criteria

1. Implement design as is above
1. Modal should appear over clicked video block.

Please also note:

1. If a conversation about the issue I’m working on is happening,
the outcomes should be documented within that issue (not in the comments).

### PR DESCRIPTION
**Pull request by Beulah-Matt**
# Description

I have implemented a video modal with a 
-Title
-Name inputs
-A thumbnail image
-Radio check marks to select video resolution.

On clicking the play button  in the video block, you are redirected to this modal.
When you click on the thumbnail, a file explorer window is opened to select a video file  

Currently, you are able to see the video in the modal in place of the thumbnail and the URL is prepopulated under the video link input.

Fixes #336
## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

#Screenshot

modal1

videoUploaded

videoInBlock

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/modals/video-block-modal/video-block-modal.component.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  styleUrls: ['./video-block-modal.component.scss']
+})
+export class VideoBlockModalComponent implements OnInit {
+
+  @Input() id: string;
+  @Input() block: VideoMessageBlock;
+  @Input() videoMessageForm: FormGroup;
+  @Input() jsPlumb: BrowserJsPlumbInstance;
+
+  modalOpen = true;
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/modals/video-block-modal/video-block-modal.component.ts:26]: Don't conditionally render this, get the reference to this dialog and close it when the close button is clicked. 

docs -> https://material.angular.io/components/dialog/overview

File: libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/modals/video-block-modal/video-block-modal.component.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      name: this.name,
+      size: this.size,
+      // VideoUrl should go here?
+      file: this.file 
+    };
+    this.applied.emit(videoBlock);
+    this.modalOpen = false
+  }
+
+  constructor(private _videoUploadService: UploadFileService,
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/video-message-block/src/lib/modals/video-block-modal/video-block-modal.component.ts:65]: We need to inject the current video data if it does exist into the dialog. Also, check the docs for more info

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Good job so far but still some updates to make 
- Some of the methods currently exist in both components, this can be better shared in a service. 
- At the moment I can select a video but I can't view it in the modal. 
- The modal is also not responsive. 
- The state of the selected does not persist into the block

**Comment by Beulah-Matt** [comment]
@LemmyMwaura @ReaganCn please confirm if the current changes are what you had in mind?

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
The state of the selected video, still doesn't persist into the blocks. and theres's also some errors in the console.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Rebase(ing) has caused you to pile on commits that were already authored and committed. Please sort this out or create a new pr with only your commits

**Pull request closed without merging**